extra/entrypoint.sh

```diff
@@ -2,20 +2,54 @@
 
 # set -e Exit the script if an error happens
 set -e
-PUID=${PUID=0}
-PGID=${PGID=0}
+
 
 files_ownership () {
-    # -h Changes the ownership of an encountered symbolic link and not that of the file or directory pointed to by the symbolic link.
-    # -R Recursively descends the specified directories
-    # -c Like verbose but report only when a change is made
-    chown -hRc "$PUID":"$PGID" /app/data
+    # Check if the /app/data folder is owned by the user invoking the container
+    if [ $(stat -c%u /app/data) !=  $(id -u) ]; then
+        echo "File ownership incorrect, attempting to fix."
+        chown -hRc "$(id -u)":"$(id -g)" /app/data || echo "ERROR: Failed to set file ownership. Please run 'sudo chown -R $(id -u):$(id -g) /path/to/container/volume' to resolve."; exit 1
+    fi
```

Comment:
We are missing echo for "success" here. Directory and file permission has this message
```suggestion
        chown -hRc "$(id -u)":"$(id -g)" /app/data || echo "ERROR: Failed to set file ownership. Please run 'sudo chown -R $(id -u):$(id -g) /path/to/container/volume' to resolve."; exit 1
        echo "File ownership fix succesful! Continuing."
    fi
```